```
======================================================================
FAIL: custom.icds.tests.test_ls_no_submissions:TestLSSubmissionPerformanceIndicator.test_form_sent_eight_days_ago
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python3.6/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/custom/icds/tests/test_ls_no_submissions.py", line 78, in test_form_sent_eight_days_ago
    self.assertTrue('one week' in message)
AssertionError: False is not true
-------------------- >> begin captured logging << --------------------
```

--- a/custom/icds/messaging/indicators.py
+++ b/custom/icds/messaging/indicators.py
@@ -293,6 +293,14 @@ class AWWVHNDSurveyIndicator(AWWIndicator):
             return []
 
 
+def _get_last_submission_dates(awc_ids):
+    return {
+        row['awc_id']: row['last_submission']
+        for row in AggregateInactiveAWW.objects.filter(
+        awc_id__in=awc_ids).values('awc_id', 'last_submission').all()
+    }
+
+
 class LSSubmissionPerformanceIndicator(LSIndicator):
     template = 'ls_no_submissions.txt'
     slug = 'ls_6'
@@ -300,11 +308,7 @@ class LSSubmissionPerformanceIndicator(LSIndicator):
     def __init__(self, domain, user):
         super(LSSubmissionPerformanceIndicator, self).__init__(domain, user)
 
-        self.last_submission_dates = {
-            row['awc_id']: row['last_submission']
-            for row in AggregateInactiveAWW.objects.filter(
-                awc_id__in=set(self.awc_locations)).values('awc_id', 'last_submission').all()
-        }
+        self.last_submission_dates = _get_last_submission_dates(awc_ids=set(self.awc_locations))
 
     def get_messages(self, language_code=None):
         messages = []

--- a/custom/icds/tests/test_ls_no_submissions.py
+++ b/custom/icds/tests/test_ls_no_submissions.py
@@ -3,7 +3,6 @@ from django.test import TestCase
 from mock import patch
 import pytz
 
-from corehq.apps.es.fake.users_fake import UserESFake
 from corehq.apps.domain.shortcuts import create_domain
 from corehq.apps.locations.tests.util import (
     LocationStructure,
@@ -16,8 +15,7 @@ from custom.icds.messaging.custom_content import run_indicator_for_user
 from custom.icds.messaging.indicators import LSSubmissionPerformanceIndicator
 
 
-@patch('corehq.apps.locations.dbaccessors.UserES', UserESFake)
-@patch('custom.icds.messaging.indicators.get_last_submission_time_for_users')
+@patch('custom.icds.messaging.indicators._get_last_submission_dates')
 class TestLSSubmissionPerformanceIndicator(TestCase):
     domain = 'domain'
 
@@ -40,11 +38,11 @@ class TestLSSubmissionPerformanceIndicator(TestCase):
         cls.loc_types = setup_location_types_with_structure(cls.domain, location_type_structure)
         cls.locs = setup_locations_with_structure(cls.domain, location_structure)
         cls.ls = cls._make_user('ls', cls.locs['LSL'])
-        cls.aww = cls._make_user('aww', cls.locs['AWC1'])
+        cls.awc1 = cls.locs['AWC1']
+        cls.awc2 = cls.locs['AWC2']
 
     @classmethod
     def tearDownClass(cls):
-        UserESFake.reset_docs()
         cls.domain_obj.delete()
         super(TestLSSubmissionPerformanceIndicator, cls).tearDownClass()
 
@@ -52,7 +50,6 @@ class TestLSSubmissionPerformanceIndicator(TestCase):
     def _make_user(cls, name, location):
         user = CommCareUser.create(cls.domain, name, 'password')
         user.set_location(location)
-        UserESFake.save_doc(user._doc)
         return user
 
     @property
@@ -61,17 +58,26 @@ class TestLSSubmissionPerformanceIndicator(TestCase):
         return datetime.now(tz=tz).date()
 
     def test_form_sent_today(self, last_sub_time):
-        last_sub_time.return_value = {self.aww.get_id: self.today}
+        last_sub_time.return_value = {
+            self.awc1.location_id: self.today,
+            self.awc2.location_id: self.today,
+        }
         messages = run_indicator_for_user(self.ls, LSSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 0)
 
     def test_form_sent_seven_days_ago(self, last_sub_time):
-        last_sub_time.return_value = {self.aww.get_id: self.today - timedelta(days=7)}
+        last_sub_time.return_value = {
+            self.awc1.location_id: self.today - timedelta(days=7),
+            self.awc2.location_id: self.today - timedelta(days=7)
+        }
         messages = run_indicator_for_user(self.ls, LSSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 0)
 
     def test_form_sent_eight_days_ago(self, last_sub_time):
-        last_sub_time.return_value = {self.aww.get_id: self.today - timedelta(days=8)}
+        last_sub_time.return_value = {
+            self.awc1.location_id: self.today - timedelta(days=8),
+            self.awc2.location_id: self.today - timedelta(days=7)
+        }
         messages = run_indicator_for_user(self.ls, LSSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 1)
         message = messages[0]
@@ -79,7 +85,10 @@ class TestLSSubmissionPerformanceIndicator(TestCase):
         self.assertTrue('AWC1' in message)
 
     def test_form_sent_thirty_days_ago(self, last_sub_time):
-        last_sub_time.return_value = {self.aww.get_id: self.today - timedelta(days=30)}
+        last_sub_time.return_value = {
+            self.awc1.location_id: self.today - timedelta(days=30),
+            self.awc2.location_id: self.today - timedelta(days=7)
+        }
         messages = run_indicator_for_user(self.ls, LSSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 1)
         message = messages[0]
@@ -88,7 +97,9 @@ class TestLSSubmissionPerformanceIndicator(TestCase):
 
     def test_form_sent_thirty_one_days_ago(self, last_sub_time):
         # last submissions only looks 30 days into past
-        last_sub_time.return_value = {}
+        last_sub_time.return_value = {
+            self.awc2.location_id: self.today - timedelta(days=7)
+        }
         messages = run_indicator_for_user(self.ls, LSSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 1)
         message = messages[0]
@@ -96,11 +107,9 @@ class TestLSSubmissionPerformanceIndicator(TestCase):
         self.assertTrue('AWC1' in message)
 
     def test_multiple_awc_eight_days_ago(self, last_sub_time):
-        aww_2 = self._make_user('aww_2', self.locs['AWC2'])
-        self.addCleanup(aww_2.delete)
         last_sub_time.return_value = {
-            self.aww.get_id: self.today - timedelta(days=8),
-            aww_2.get_id: self.today - timedelta(days=8)
+            self.awc1.location_id: self.today - timedelta(days=8),
+            self.awc2.location_id: self.today - timedelta(days=8)
         }
         messages = run_indicator_for_user(self.ls, LSSubmissionPerformanceIndicator, language_code='en')
         self.assertEqual(len(messages), 1)

--- a/custom/icds/tests/test_ls_vhnd_survey.py
+++ b/custom/icds/tests/test_ls_vhnd_survey.py
@@ -74,7 +74,8 @@ class TestLSVHNDSurveyIndicator(TestCase, VHNDIndicatorTestMixin):
         cls.loc_types = setup_location_types_with_structure(cls.domain, location_type_structure)
         cls.locs = setup_locations_with_structure(cls.domain, location_structure)
         cls.ls = cls._make_user('ls', cls.locs['LSL'])
-        cls.aww = cls._make_user('aww', cls.locs['AWC1'])
+        cls.aww1 = cls._make_user('aww1', cls.locs['AWC1'])
+        cls.aww2 = cls._make_user('aww2', cls.locs['AWC2'])
         cls.setup_datasource()
 
     @classmethod
@@ -96,27 +97,27 @@ class TestLSVHNDSurveyIndicator(TestCase, VHNDIndicatorTestMixin):
         return datetime.now(tz=tz).date()
 
     def test_survey_date_today(self):
-        self._save_form(self.aww.get_id, self.today)
+        self._save_form(self.aww1.get_id, self.today)
+        self._save_form(self.aww2.get_id, self.today)
         messages = run_indicator_for_user(self.ls, LSVHNDSurveyIndicator, language_code='en')
         self.assertEqual(len(messages), 0)
 
     def test_form_sent_thirty_six_days_ago(self):
-        self._save_form(self.aww.get_id, self.today - timedelta(days=36))
+        self._save_form(self.aww1.get_id, self.today - timedelta(days=36))
+        self._save_form(self.aww2.get_id, self.today - timedelta(days=36))
         messages = run_indicator_for_user(self.ls, LSVHNDSurveyIndicator, language_code='en')
         self.assertEqual(len(messages), 0)
 
     def test_form_sent_thirty_seven_days_ago(self):
-        self._save_form(self.aww.get_id, self.today - timedelta(days=37))
+        self._save_form(self.aww1.get_id, self.today - timedelta(days=37))
+        self._save_form(self.aww2.get_id, self.today - timedelta(days=36))
         messages = run_indicator_for_user(self.ls, LSVHNDSurveyIndicator, language_code='en')
         self.assertEqual(len(messages), 1)
         self.assertTrue('AWC1' in messages[0])
 
     def test_multiple_locations(self):
-        aww_2 = self._make_user('aww_2', self.locs['AWC2'])
-        self.addCleanup(UserESFake.remove_doc, aww_2.get_id)
-        self.addCleanup(aww_2.delete)
-        self._save_form(self.aww.get_id, self.today - timedelta(days=37))
-        self._save_form(aww_2.get_id, self.today - timedelta(days=37))
+        self._save_form(self.aww1.get_id, self.today - timedelta(days=37))
+        self._save_form(self.aww2.get_id, self.today - timedelta(days=37))
 
         messages = run_indicator_for_user(self.ls, LSVHNDSurveyIndicator, language_code='en')
         self.assertEqual(len(messages), 1)
@@ -125,11 +126,8 @@ class TestLSVHNDSurveyIndicator(TestCase, VHNDIndicatorTestMixin):
         self.assertTrue('AWC2' in message)
 
     def test_multiple_locations_one_is_old(self):
-        aww_2 = self._make_user('aww_2', self.locs['AWC2'])
-        self.addCleanup(UserESFake.remove_doc, aww_2.get_id)
-        self.addCleanup(aww_2.delete)
-        self._save_form(aww_2.get_id, self.today - timedelta(days=37))
-        self._save_form(self.aww.get_id, self.today - timedelta(days=15))
+        self._save_form(self.aww2.get_id, self.today - timedelta(days=37))
+        self._save_form(self.aww1.get_id, self.today - timedelta(days=15))
 
         messages = run_indicator_for_user(self.ls, LSVHNDSurveyIndicator, language_code='en')
         self.assertEqual(len(messages), 1)

